content/library/advanced-features/dataframes.md

```diff
@@ -228,298 +224,88 @@ st.data_editor({
+You can use `st.data_editor` to collect tabular input from a user. When starting from an empty dataframe, default column types are text. Use column configuration to specify the data types you want to collect from users.
 
 ```python
+import streamlit as st
 import pandas as pd
 
-# create a dataframe with a boolean column
-df = pd.DataFrame({"A": [True, False, True, False]})
+df = pd.DataFrame(columns=['name','age','color'])
+colors = ['red', 'orange', 'yellow', 'green', 'blue', 'indigo', 'violet']
+config = {
+    'name' : st.column_config.TextColumn('Full Name (required)', width='large', required=True),
+    'age' : st.column_config.NumberColumn('Age (years)', min_value=0, max_value=122),
+    'color' : st.column_config.SelectboxColumn('Favorite Color', options=colors)
+}
 
-# show the dataframe with checkboxes
-st.dataframe(df)
-```
-
-![data-editor-dataframe-boolean.gif](/images//data-editor-dataframe-boolean.gif)
-
-Notice you cannot change their values from the frontend. To let users check and uncheck values, we display the dataframe with `st.data_editor` instead:
+result = st.data_editor(df, column_config = config, num_rows='dynamic')
 
-```python
-import pandas as pd
-
-# create a dataframe with a boolean column
-df = pd.DataFrame({"A": [True, False, True, False]})
-
-# show the data editor with checkboxes
-st.data_editor(df)
+if st.button('Get results'):
+    st.write(result)
 ```
 
-![data-editor-boolean.gif](/images/data-editor-boolean.gif)
-
-### Categorical columns (selectboxes)
-
-To render columns as selectboxes with `st.data_editor`, set the type of the Pandas column as `category`:
-
-```python
-import pandas as pd
-
-df = pd.DataFrame(
-    {"command": ["st.selectbox", "st.slider", "st.balloons", "st.time_input"]}
-)
-df["command"] = df["command"].astype("category")
-
-edited_df = st.data_editor(df)
-```
+<Cloud src="https://doc-column-config-empty.streamlit.app/?embed=true" height="300"/>
 
-In some cases, you may want users to select categories that aren’t in the original Pandas DataFrame. Let’s say we use `df` from above. Currently, the `command` column can take on four unique values. What should we do if we want users to see additional options such as `st.button` and `st.radio`?
+<Important>
 
-To add additional categories to the selection, use [pandas.Series.cat.add_categories](https://pandas.pydata.org/docs/reference/api/pandas.Series.cat.add_categories.html):
+If you use `pandas<2.0` and want to hide the index for an empty dataframe with dynamic rows, you will need to manually declare a `pandas.RangeIndex`. Without doing so, the first column of the of the data editor will become an input column for the index. New rows won't be created if this is hidden. You can manually declare an empty dataframe's index like this:
```

Comment:
That seems super edge casey, not sure if we need to keep this in here. Maybe we could also move it to KB? Or document it on the API reference for st.data_editor? Or just leave it out altogether, given that more and more people will switch to pandas 2 eventually, anyway?

Reply:
I think if this is an issue people have, they will just land on a forum post for this anyway...

Reply:
I'll remove it, but also @LukasMasuch since he raised the issue originally.